Add paste service in `common.py`

janaSunrise opened the pull request.

janaSunrise committed:
Add paste.

ItsDrike commented on cogs/common.py:
````diff
@@ -123,6 +125,20 @@ async def thanks(self, ctx: Context, member: Member, *, reason: str = None) -> N
+
+            await ctx.send(f"FILE PASTE : {file_paste}")
+
+    def _clean_code(self, code: str) -> str:
+        if code.startswith("```") and code.endswith("```"):
+            return "\n".join(code.split("\n")[1:-1])
+        return code.strip("\n")
````
This wouldn't work on something like <code>\`\`\`this\`\`\`</code> where the code is on a single line but wrapped in a codeblock (<code>```</code>)

janaSunrise replied:
i know.

ItsDrike replied:
so, fix it?

ItsDrike reviewed (changes requested):
Looks fine except the `_clean_code` function as pointed out below, but before merging we'll need to make sure the new `paste` command works properly.

ItsDrike committed:
Merge branch 'master' into paste

ItsDrike committed:
Add regex matching for `_clean_code`

- Match code with regex instead of only using code from second to last
  lines if ``` is in code, which doesn't make sense
- Add regex patterns to recognise both regular codeblocks and simple
  code highlights

ItsDrike committed:
Merge branch 'master' into paste

ItsDrike commented on bot/cogs/common.py:
```diff
@@ -169,6 +165,30 @@ async def thanks(self, ctx: Context, member: Member, *, reason: str = None) -> N
+            key = (await resp.json())['key']
+            file_paste = 'https://www.hasteb.in/' + key
+
+            await ctx.send(f"FILE PASTE : {file_paste}")
```
Consider using an embed here

janaSunrise committed:
Apply requested changes

janaSunrise committed:
Merge branch 'master' into paste
